packages/host/app/services/operator-mode-state-service.ts

```diff
@@ -112,7 +112,42 @@ export default class OperatorModeStateService extends Service {
     for (let item of stackItems) {
       if ('card' in item && item.card.id == id) {
         let document = await this.cardService.serializeCard(item.card);
-        document.data = merge(document.data, patch);
+        if (patch.attributes) {
+          for (let key of Object.keys(patch.attributes)) {
+            if (!(key in item.card)) {
+              throw new Error(`Attribute "${key}" does not exist in card`);
+            }
+          }
+          document.data.attributes = merge(
+            document.data.attributes,
+            patch.attributes,
+          );
+        }
+
+        if (patch.relationships) {
+          for (let [key, value] of Object.entries(patch.relationships)) {
+            if (!Array.isArray(value)) {
+              continue;
+            }
+            if (!(key in item.card)) {
+              throw new Error(`Relationship "${key}" does not exist in card`);
+            }
+            if (document.data.relationships) {
+              Object.entries(document.data.relationships)
+                .filter(([fieldName]) => fieldName.includes('.'))
+                .map(([fieldName]) => {
+                  if (fieldName.split('.')[0] === key) {
+                    delete document.data.relationships![fieldName];
+                  }
+                });
+            }
+          }
+          document.data.relationships = merge(
```

Comment:
on the server we are also using `mergeWith` for relationship patching as well, in fact there is a runtime-common function for merging relationships that we use from teh server, `@cardstack/runtime-common/merge-relationships`, is there a reason why we wouldn't use that here as well?

Reply:
Made the updates to use `mergeWith` and `mergeRelationships`. Now this works much better.